device/thunder_ripple_sdk/src/client/thunder_async_client.rs

```diff
@@ -194,37 +194,46 @@ impl ThunderAsyncClient {
                 .generate_plugin_activation_request(callsign.clone());
             return Ok(request.to_string());
         }
+
         // Generate the appropriate JSON-RPC request based on the type of DeviceChannelRequest
```

Comment:
RELATED: We seem to do a lot of en/decoding of request data across different types that get queued, we should take another look and see if that can get cut-down a little, doesn't seem like we're storing things in the most ideal format? Not now but something for later...